lib/NetworkManager.js

```diff
@@ -341,6 +341,49 @@ class Request {
   }
 
   /**
+   * @param {!Object} response
+   */
+  async respond(response = {}) {
+    // DataURL's are not interceptable. In this case, do nothing.
+    if (this.url.startsWith('data:'))
+      return;
+    console.assert(this._allowInterception, 'Request Interception is not enabled!');
+    console.assert(!this._interceptionHandled, 'Request is already handled!');
+    this._interceptionHandled = true;
+    const responseLines = [];
+    const statusCode = response.status || 200;
+    const statusText = statusTexts[statusCode] || '';
+    responseLines.push(`HTTP/1.1 ${statusCode} ${statusText}`);
+    const headers = {};
+    if (response.headers) {
+      for (const header of Object.keys(response.headers))
+        headers[header.toLowerCase()] = response.headers[header];
+    }
+    if (response.body.length && !('content-length' in headers))
+      headers['content-length'] = response.body.length;
```

Comment:
@aslushnikov Do you want to perhaps use `Buffer.byteLength(body)` instead of `body.length`?

Reply:
@paambaati this is reasonable, thanks.